code/modules/mob/living/life.dm

```diff
@@ -47,6 +47,25 @@
 /mob/living/proc/handle_chemicals_in_body()
 	SHOULD_CALL_PARENT(TRUE)
 	chem_effects = null
+	if(status_flags & GODMODE)
+		return FALSE
+	if(isSynthetic())
+		return FALSE
+	var/datum/reagents/metabolism/ingested_reagents = get_ingested_reagents()
```

Comment:
idk if there's point in getting it here if you check in metabolize_ingested_reagents() anyway

Reply:
I ended up changing that block to retrieve the datums via `metabolize_x_reagent()` because it then checks them for `chem_dose` updates.